server/src/test/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -504,6 +604,26 @@ public void testAdditionalSettingsClash() {
         assertTrue(msg, msg.contains("plugin [" + AdditionalSettingsPlugin2.class.getName()));
     }
 
+    public void testPluginComponentInterfaceBinding() throws IOException, NodeValidationException {
+        List<Class<? extends Plugin>> plugins = basePlugins();
+        plugins.add(MockPluginWithAltImpl.class);
+        try (Node node = new MockNode(baseSettings().build(), plugins)) {
+            MockPluginWithAltImpl.MyInterface myInterface = node.injector().getInstance(MockPluginWithAltImpl.MyInterface.class);
+            MockPluginWithAltImpl plugin = node.getPluginsService().filterPlugins(MockPluginWithAltImpl.class).get(0);
+            if (plugin.getRandomBool()) {
+                assertTrue(myInterface instanceof MockPluginWithAltImpl.Foo);
```

Comment:
nit: Can we use assertThat for these? The output is nicer when they do fail, eg with instanceOf it will tell what hte actual class is.

Reply:
done [`e7ef710` (#94347)](https://github.com/elastic/elasticsearch/pull/94347/commits/e7ef7109c17540a9a4aaf7d9b637d6a98905b1d5) (I either forgot or never knew that existed, thanks :))